### PR TITLE
Services: Add service to finder contextual menu.

### PR DESCRIPTION
**Pull request by lolgear**
- [x] #287

### CODE REVIEW

**Comment by lolgear** [comment]
@lucasderraugh Awaiting for review :)

File: GitUp/Application/ServicesProvider.m [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+// 5.3. /System/Library/CoreServices/pbs -dump_cache
+// 6. Be sure that your .app is appearing in dump_cache output.
+// 7. Check finder contextual menu.
+- (void)openRepository:(NSPasteboard *)pasteboard userData:(NSString *)userData error:(NSError * __autoreleasing *)error {
+  if ([self canOpenItem:pasteboard]) {
+    NSURL *url = [self items:pasteboard].firstObject;
+    [self.appDelegate _openRepositoryWithURL:url withCloneMode:kCloneMode_None windowModeID:NSNotFound];
+  }
+  else {
+    // fill error if needed.
```
Comment (lucasderraugh) [inline GitUp/Application/ServicesProvider.m]: Can we report back if the item doesn't contain a git repository?

Comment (lolgear) [reply GitUp/Application/ServicesProvider.m]: @lucasderraugh Sure.
Does this check exist in `GCRepository`?

**Comment by lolgear** [comment]
@lucasderraugh 
Strange build failure. GenerateDsym fails for iPhone target. Hmm...

**Comment by lolgear** [comment]
@lucasderraugh 
Ah, I figured it out. Xcode refuses to skip warnings.
Could you check and merge it?

**Comment by lolgear** [comment]
Ready?

**Comment by lolgear** [comment]
@lucasderraugh Ready?

**Comment by lucasderraugh** [comment]
Looks good. Sorry for taking forever to review this 😬.

**Pull request merged**